src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -904,29 +942,228 @@ void BrgemmEmitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brgKernel, in
     // `host_isa::vecs_count`.
     h->sub(h->rsp, get_max_vecs_count() * get_vec_length());
     for (size_t i = 0; i < get_max_vecs_count(); ++i)
-        h->uni_vmovups(h->ptr[h->rsp + i * get_vec_length()], Vmm(i));
+        h->uni_vmovups(h->ptr[h->rsp + i * get_vec_length()], Zmm(i));
+
+    kernel_call(brg_kernel, regs, offsets);
+
+    // restore vector registers
+    for (int i = static_cast<int>(get_max_vecs_count()) - 1; i >= 0; --i) {
+        h->uni_vmovups(Zmm(i), h->ptr[h->rsp + i * get_vec_length()]);
+    }
+    h->add(h->rsp, (get_max_vecs_count()) * get_vec_length());
+
+    // restore k registers
+    for (int i = n_k_regs_to_save - 1; i >= 0; --i) {
+        if (mayiuse(avx512_core))
+            h->kmovq(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
+        else
+            h->kmovw(Opmask(i), h->ptr[h->rsp + i * k_mask_size]);
+    }
+    h->add(h->rsp, n_k_regs_to_save * k_mask_size);
+
+    // restore gpr registers
+    for (int i = n_gprs_to_save - 1; i >= 0; --i)
+        h->mov(gprs_to_save[i], h->ptr[h->rsp + i * gpr_size]);
+    h->add(h->rsp, n_gprs_to_save * gpr_size);
+}
+
+void BrgemmBaseEmitter::kernel_execute(const brgemm_kernel_t *brg_kernel,
+                                       const void *A, const void *B, void *C, void *scratch, int with_comp) {
+    brgemm_kernel_params_t brgemm_p;
 
+    brgemm_p.batch = nullptr;  // default value
+    brgemm_p.ptr_A = A;
+    brgemm_p.ptr_B = B;
+    brgemm_p.ptr_C = C;
+    brgemm_p.ptr_D = C;
+    brgemm_p.ptr_buf = scratch;
+    brgemm_p.ptr_bias = nullptr;
+    brgemm_p.do_post_ops = with_comp;
+    brgemm_p.do_apply_comp = with_comp;
+    brgemm_p.skip_accm = 0;
+    brgemm_p.BS = 1;  // default value
+    assert(brg_kernel);
+    (*brg_kernel)(&brgemm_p);
+}
+
+BrgemmEmitter::BrgemmEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa, const std::shared_ptr<ov::Node>& n)
+    : BrgemmBaseEmitter(h, isa, n) {}
+
+std::vector<size_t> BrgemmEmitter::init_kernel_offsets(size_t mb, size_t M_blk, size_t LDA, size_t LDC,
+                                                       size_t k, size_t K0_step0, size_t K0_step1,
+                                                       size_t n, size_t N0_step0, size_t N0_step1) const {
+    const size_t in0_offset = load_offset_a + (k * K0_step0 + mb * M_blk * LDA) * io_data_size[0];
+    const size_t in1_offset = load_offset_b + (k * K0_step1 + n * N0_step0) * io_data_size[1];
+    const size_t out0_offset = store_offset_c + (n * N0_step1 + mb * M_blk * LDC) * io_data_size[2];
+    return { in0_offset, in1_offset, out0_offset };
+}
+
+void BrgemmEmitter::kernel_call(const brgemm_kernel_t *brg_kernel,
```

Comment:
Code of this emitter is 99.32% identical to the `BrgemmWithScratchEmitter::kernel_call`. Can we generalize them to avoid repetition? It seems at he first glance that we just need to insert a couple of `if(regs.size() == 3)`